index.js

```diff
@@ -88,33 +119,71 @@ class FilteredPage extends React.Component {
       )
   }
   handleSelectChange(e) {
-    fetch(apiURL + e.target.value + '.json' + apiKEY + this.state.apiInput)
-      .then(res => res.json())
-      .then(res =>
-        this.setState({
-          result: res.results
-        }))
+    if (e.target.value === 'saved') {
+      this.setState({
+        result: []
+      })
+    }
+    else {
+      fetch(apiURL + e.target.value + '.json' + apiKEY + this.state.apiInput)
+        .then(res => res.json())
+        .then(res =>
+          this.setState({
+            result: res.results
+          }))
+    }
+  }
+  handleQueue(result) {
+    myStorage.setItem(result.title, JSON.stringify(result))
+    this.setState({
+      saved: Object.keys(myStorage).map(result =>
+        JSON.parse(localStorage.getItem(result))
+      )
+    })
+  }
+  handleRemove(result) {
+    myStorage.removeItem(result.title)
+    this.setState({
+      saved: Object.keys(myStorage).map(result =>
+        JSON.parse(localStorage.getItem(result))
+      )
+    })
+  }
+  componentDidMount() {
+    this.setState({
+      saved: Object.keys(myStorage).map(result =>
+        JSON.parse(localStorage.getItem(result))
```

Comment:
This is a little confusing. Are `myStorage` and `localStorage` the same thing? It might be cleaner to just avoid creating the `myStorage` variable in the first place.